packages/blade/src/tokens/theme/overrideTheme.stories.tsx

```diff
@@ -0,0 +1,196 @@
+import type { Meta } from '@storybook/react';
+import { Title, Subtitle, Primary, ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
+import type { ReactElement } from 'react';
+import outdent from 'outdent';
+import { Highlight, Link } from '@storybook/design-system';
+import overrideTheme from './overrideTheme';
+import paymentTheme from './paymentTheme';
+import { Button } from '~components/Button';
+import { BladeProvider } from '~components/BladeProvider';
+import { Checkbox, CheckboxGroup } from '~components/Checkbox';
+import Box from '~components/Box';
+import { Text, Title as BladeTitle } from '~components/Typography';
+
+const Page = (): ReactElement => {
+  return (
+    <>
+      <Title />
+      <Subtitle>
+        overrideTheme takes two themes, baseThemeTokens & overrides and returns a new ThemeTokens
+        object, which you can pass into BladeProvider.
+      </Subtitle>
+      <Link href="https://github.com/razorpay/blade/blob/master/packages/blade/src/tokens/theme/overrideTheme.stories.tsx">
+        Link to source example
+      </Link>
+      <br />
+      <br />
+      <Title>Usage</Title>
+      <Highlight language="tsx">{outdent`
+        const customTheme = overrideTheme({
+          baseThemeTokens: paymentTheme, // theme to override
+          overrides: {
+            colors: {
+              onLight: {
+                brand: {
+                  primary: {
+                    '500': 'red',
+                  },
+                },
+              },
+            },
+          },
+        });
+
+        const App = () => {
+          return &lt;BladeProvider themeTokens={customTheme} \/\>
+        }
+        `}</Highlight>
+      <Title>Example</Title>
+      <Primary />
+      <Title>Properties</Title>
+      <ArgsTable story={PRIMARY_STORY} />
+      <Stories />
+    </>
+  );
+};
+
+export default {
+  title: 'Utilities/overrideTheme',
+  args: {
+    hue: 28,
+  },
+  argTypes: {
+    hue: {
+      control: {
+        name: 'hue',
+        type: 'range',
+        min: 0,
+        max: 360,
+        step: 1,
+      },
+      description: 'Change this hue to see the colors change',
+    },
+  },
+  parameters: {
+    docs: {
+      page: Page,
+    },
+  },
+} as Meta;
+
+const OverrideThemeExample = (props: { hue: number }): React.ReactElement => {
+  // Global parameters
+  const hue = props.hue;
+  const primaryColor = `hsla(${hue}, 100%, 50%, 1)`;
+  const secondaryColor = `hsla(${hue}, 100%, 99%, 1)`;
+  const tertiaryColor = `hsla(${hue}, 100%, 96%, 1)`;
```

Comment:
Would it be also possible to create `primaryHoverColor`, `primaryActiveColor`, etc? Right now there are no hover and active states in the example. We shouldn't encourage people to override without hover & active states

Reply:
Yeah should be possible.

Reply:
Added